model_analyzer/config/generate/neighborhood.py

```diff
@@ -71,31 +73,81 @@ def enough_coordinates_initialized(self) -> bool:
         """
         Returns true if enough coordinates inside of the neighborhood
         have been initialized. Else false
+
+        If the neighborhood is in slow mode, this means all next door neighbors
+        must be visited
+        """
+        if self._is_slow_mode():
+            return self._are_all_next_door_neighbors_visited()
+        else:
+            min_initialized = self._config.get_min_initialized()
+            num_initialized = len(self._get_initialized_coordinates())
+            return num_initialized >= min_initialized
+
+    def force_slow_mode(self):
         """
-        min_initialized = self._config.get_min_initialized()
-        num_initialized = len(self._get_initialized_coordinates())
-        return num_initialized >= min_initialized
+        When called, forces the neighborhood into slow mode
+        """
+        self._force_slow_mode = True
 
     def calculate_new_coordinate(self,
                                  magnitude: int,
                                  enable_clipping: bool = True,
                                  clip_value: int = 2) -> Coordinate:
         """
         Based on the measurements in the neighborhood, determine where
-        the next location should be
+        the next location should be.
+
+        If the neighborhood is in slow mode, return the best found measurement
+        Otherwise calculate a new coordinate from the measurements
 
         Parameters
         ----------
         magnitude
             How large of a step to take
-        disable_clipping
+        enable_clipping
             Determines whether or not to clip the final step vector.
+        clip_value
+            What value to clip the vector at, if it is enabled
 
         Returns
         -------
         new_coordinate
             The new coordinate computed based on the neighborhood measurements.
         """
+
+        if self._is_slow_mode():
+            return self._get_best_coordinate_found()
+        else:
+            return self._calculate_new_coordinate(magnitude, enable_clipping,
+                                                  clip_value)
+
+    def _get_best_coordinate_found(self) -> Coordinate:
+        vectors, measurements = self._get_measurements_passing_constraints()
+
+        if len(vectors) == 0:
+            return self._home_coordinate
+
+        home_measurement = self._get_home_measurement()
+
+        if home_measurement.is_passing_constraints():
+            vectors.append(Coordinate([0] * self._config.get_num_dimensions()))
+            measurements.append(home_measurement)
+
+        best_vector = vectors[0]
+        best_measurement = measurements[0]
+
+        for v, m in zip(vectors, measurements):
```

Comment:
Can't you just make this a tuple and then call sort?

Reply:
Lines 137-143 can be replaced with one line: 
best_vector, best_measurement = zip(vectors, measurements).sort()[-1]

(or you can change the sort to be descending so that best is the first element, if that is more readable)

Reply:
The zip has to be measurements first (since that is what we want to sort on, or else we need to do extra stuff to tell it to sort differently)

Zip doesn't have sort, but sorted() supports it.

Thus, the line is 
```
        _, best_vector = sorted(zip(measurements, vectors))[-1]
```

I have updated it to that, and confirmed the results of the entire sweep are still exactly the same